test/pubsub.ts

```diff
@@ -282,6 +282,23 @@ describe('PubSub', () => {
     it('should localize a Promise override', () => {
       assert.strictEqual(pubsub.Promise, OPTIONS.promise);
     });
+
+    describe('with PUBSUB_EMULATOR_HOST & GOOGLE_CLOUD_PROJECT environment vars', () => {
+      before(() => {
+        process.env.PUBSUB_EMULATOR_HOST = 'localhost:9090';
```

Comment:
Assuming these env vars are set when some one runs the test these hooks would remove them entirely. It would be good to cache the current values and in the after hook reset them to said values.

Reply:
I copied this code from below. I'm fine with adding it.

In general I would make the assumption that each test (or its registered `before`/`beforeAll`) sets up the whole environment. Otherwise your tests rely on a specific order of execution.

Also keep in mind that (re-)setting them to `undefined` is not the same as just calling `delete`.

Reply:
What should I do here? Add the reset?